### PR TITLE
Test Poseidon

### PR DESCRIPTION
**Pull request by lazovicff**

### CODE REVIEW

**Comment by lazovicff** [comment]
Hmm, this compiles locally

**Comment by lazovicff** [comment]
There are some GitHub Actions errors, but in the meantime it can be reviewed @drewstone

File: src/crh/poseidon/mod.rs [hunk 1]
```diff
@@ -31,191 +30,209 @@ pub trait PoseidonRoundParams<F: PrimeField>: Default + Clone {
...
+
+        if (input.len() * 8) != Self::INPUT_SIZE_BITS * P::WIDTH {
+            panic!(
+                "incorrect input length {:?} for width {:?}",
+                input.len() / 32,
+                P::WIDTH,
+            );
+        }
+
+        // Not giving expected results
```
Comment (drewstone) [inline src/crh/poseidon/mod.rs:155]:
```suggestion
```

File: src/crh/poseidon/mod.rs [hunk 2]
```diff
@@ -31,191 +30,209 @@ pub trait PoseidonRoundParams<F: PrimeField>: Default + Clone {
...
+        // let elts: Vec<F> = input.to_field_elements().unwrap_or(Vec::new());
```
Comment (drewstone) [inline src/crh/poseidon/mod.rs:156]:
```suggestion
```

**Review by drewstone — APPROVED** [review]
Should remove some excess comments but otherwise looks great!

File: src/crh/poseidon/mod.rs (hunk 1 above)
Comment (lazovicff) [reply src/crh/poseidon/mod.rs:155]: I left those comments to initiate a discussion about it. Seems like `to_field_elements` is the right way to do it, but I'm not sure why it's not giving correct results

Comment (lazovicff) [reply src/crh/poseidon/mod.rs:155]: As well as `to_constraint_field` for getting `FpVar`

**Comment by drewstone** [comment]
Still needs some style fixes.

**Comment by lazovicff** [comment]
For now, it's best to run `cargo fmt`, but long term its best to add `rustfmt.toml`

File: src/crh/poseidon/mod.rs (hunk 1 above)
Comment (drewstone) [reply src/crh/poseidon/mod.rs:155]: Ah ok I missed this. So we should add it back? Also then are the constraints unimplemented against these new changes?

**Comment by drewstone** [comment]
Can we get a tally of what's left @filiplazovic ? Including if we are still missing new implementations of the constraints.

**Pull request merged**